components/widgets/menu/menu_wrapper.tsx

```diff
@@ -77,12 +78,23 @@ export default class MenuWrapper extends React.PureComponent<Props, State> {
         }
     }
 
-    private toggle = () => {
-        const newState = !this.state.open;
-        this.setState({open: newState});
-        if (this.props.onToggle) {
-            this.props.onToggle(newState);
+    toggle = (e: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
+        /**
+         * This is only here so that we can toggle the menus in the sidebar, because the default behavior of the mobile
+         * version (ie the one that uses a modal) needs propagation to close the modal after selecting something
+         * We need to refactor this so that the modal is explicitly closed on toggle, but for now I am aiming to preserve the existing logic
+         * so as to not break other things
+        **/
+        if (this.props.stopPropagationOnToggle) {
+            e.preventDefault();
+            e.stopPropagation();
         }
+        const newState = !this.state.open;
+        this.setState({open: newState}, () => {
```

Comment:
Why does this need to wait until after the state has been updated to call `onToggle`?

Reply:
I don't think there's a requirement to, is this a bad pattern or something?

Reply:
It's not a bad pattern to use it when it's needed, but we shouldn't use it unless it's actually necessary since it complicates code flow and may cause difficult to understand bugs like wrapping the callback in a `setTimeout(..., 0)` would.

Reply:
Sure, I can remove it then

Reply:
Looked into it, it actually is required so that the menu renders as open and then the `SidebarMenu` can move the menu to the correct position.